source/kernel/src/drivers/serial_mux.rs

```diff
@@ -28,40 +41,154 @@ pub enum Response {
     PortRegistered(PortHandle),
 }
 
-struct Commander {
-    cmd: KConsumer<Message>,
+struct PortInfo {
+    port: u16,
+    upstream: bbq::SpscProducer,
+}
+
+struct MuxingInfo {
+    kernel: &'static Kernel,
+    ports: HeapFixedVec<PortInfo>,
+    max_frame: usize,
+}
+
+struct CommanderTask {
+    cmd: KConsumer<Message<SerialMux>>,
     out: bbq::MpscProducer,
-    mux: HeapArc<Mutex<SerialMux>>,
+    mux: HeapArc<Mutex<MuxingInfo>>,
 }
 
-impl Commander {
-    async fn run(self) {
-        loop {
-            let msg = self.cmd.dequeue_async().await.unwrap();
-            let Message { req, resp } = msg;
-            match req {
-                Request::RegisterPort { port_id, capacity } => {
-                    let res = {
-                        let mut mux = self.mux.lock().await;
-                        mux.register_port(port_id, capacity, &self.out).await
-                    };
-                    resp.enqueue_async(res.map(|ph| Response::PortRegistered(ph)))
-                        .await
-                        .map_err(drop)
-                        .unwrap();
-                }
-            }
+struct IncomingMuxerTask {
+    buf: HeapArray<u8>,
+    idx: usize,
+    incoming: bbq::Consumer,
+    mux: HeapArc<Mutex<MuxingInfo>>,
+}
+
+// impl SerialMux
+
+impl RegisteredDriver for SerialMux {
+    type Request = Request;
+    type Response = Response;
+    const UUID: Uuid = uuid!("54c983fa-736f-4223-b90d-c4360a308647");
```

Comment:
nit: it would probably be good if the UUID in the `registry` module's list of well-known UUIDs referenced this constant, instead of the same string occurring in two places...